pkg/downloader/manager.go

```diff
@@ -247,7 +247,11 @@ func (m *Manager) downloadAll(deps []*chart.Dependency) error {
 	}
 
 	destPath := filepath.Join(m.ChartPath, "charts")
-	tmpPath := filepath.Join(m.ChartPath, "tmpcharts")
+	tmpPath, err := ioutil.TempDir("", "tmpcharts-*")
+	if err != nil {
+		return errors.Wrap(err, "failed to create tmpcharts dir")
+	}
+	os.Remove(tmpPath)
```

Comment:
If I understand this correctly, we create a `tmpcharts-*`directory via `ioutil.TempDir` and then delete it, so that we can rename `charts` to it later.

Isn't this unnecessarily brittle? 
`os.Remove` [can return an error](https://golang.org/src/os/file_unix.go?s=9160:9190#L324) in which case we might (or might not) fail a bit further down.

Would it not be cleaner to just generate a name (something like what is being done inside `ioutil.TempDir`), make sure it doesn't exist via `os.Stat` and move on from there.

Reply:
Having said that, I have no authority here 🙃

Reply:
I personally prefer to do as little work as possible and rely on libraries rather than copy+pasting code.

That said, I'm happy to do a certain amount of extra work as requested by an org in order to get a PR merged.

I could also use `ioutil.Tempfile` and just get the filename (minus the path) and then use that as the local name instead.

That seems like a reasonable middle ground. Yes, crossing file systems is something one should generally try to avoid.

Reply:
> I personally prefer to do as little work as possible and rely on libraries rather than copy+pasting code.

Amen to that, and on second thought, I think the touch/delete on the dir is not that big a deal.

How about something like this then? (https://github.com/gzur/helm/commit/ca0fe09bbccb567273563243cf07c14e80f656d3)
```
tmpPath, err := ioutil.TempDir(m.ChartPath, "tmpcharts-*")
```

Reply:
@gzur I've pulled in your changes and fixed them up. If you don't care about authorship, I can squash it into one of the other commits. And/or I can add a `co-authored-by` field

Reply:
I was gonna be all blasé and tell you just squash and ship, but on second thought a mention in the `co-authored-by` field would be nice.

Reply:
Thinking about it, I think I want to squash and use co-authored-by, because even the initial suggestion was yours.

Reply:
Sigh, your changes are also to the test by @deathname. So, either I fold everything together, or leave things as is.

I'm leaving them alone unless someone directs otherwise.

Reply:
How do we move this forward?